File: frontend/lib/models/chat_message.dart
// 채팅 메세지 모델

class ChatMessage {
  final String text;
  final bool isUser; // 사용자 or 챗봇
  final String time;
  final String date;
  // 위치 정보는 nullable
  //37.610837, 126.996379
  final double? lat;
  final double? lng;
  final int roomId; //채팅방 ID int형으로
  final String? placeName;

  ChatMessage({
    required this.text,
    required this.isUser,
    required this.time,
    required this.date,
    this.lat,
    this.lng,
    required this.roomId,
    this.placeName,
  });
}

File: backend/src/main/java/com/jeongneung/JeongneungChingu/domain/entity/ChatMessage.java
package com.jeongneung.JeongneungChingu.domain.entity;



import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private User user;

    private String text;

    private boolean isUser;

    private String placeName;

    public boolean isUser() {      // ChatService가 호출할 메서드
        return isUser;
    }
    private String time;
    private String date;
    private Double lat;
    private Double lng;
    private int roomId;
}